src/stream/callback.rs

```diff
@@ -1,17 +1,27 @@
 use std::fmt;
 use std::future::Future;
-use std::path::PathBuf;
+use std::path::{PathBuf, Path};
 use std::pin::Pin;
 
 use futures::FutureExt;
 use tokio::sync::{mpsc::{Receiver, Sender}, Mutex};
 use tokio::sync::mpsc;
 
+use crate::Result;
+
 pub type OnProgressClosure = Box<dyn Fn(CallbackArguments)>;
 pub type OnProgressAsyncClosure = Box<dyn Fn(CallbackArguments) -> Pin<Box<dyn Future<Output=()>>>>;
 pub type OnCompleteClosure = Box<dyn Fn(Option<PathBuf>)>;
 pub type OnCompleteAsyncClosure = Box<dyn Fn(Option<PathBuf>) -> Pin<Box<dyn Future<Output=()>>>>;
 
+#[derive(Debug)]
+pub(crate) enum InternalSignals {
```

Comment:
An instance of this `enum` probably represents an `InternalSignal` rather than `InternalSignals`.